base/steer/FairRootManager.h

```diff
@@ -421,7 +417,7 @@ class FairRootManager : public TObject
     // data members
     Int_t fId;   // This manager ID
 
-    ClassDef(FairRootManager, 12);
+    ClassDef(FairRootManager, 13);
```

Comment:
This inherits from `TObject`. Can you please use `ClassDefOverride` here, if you touch this line anyway?

Reply:
Why this should be changed? We don't use the override keyword in the class. According to the ROOT team this macro is needed when this is used

> ClassDefOverride: Use this one if your class definition use the c++ `override` keyword for virtual functions (If you use the regular ClassDef, the compiler might warn that the function declared in the ClassDef are missing the override keyword).

Reply:
After [looking it up](https://github.com/root-project/root/blob/0e0154b4960c92ae62df652b5b21f7d24a1fab1d/core/base/inc/Rtypes.h#L309), I agree with @fuhlig1. The difference between `ClassDef` and `ClassDefOverride` seems to be that the generated code literally uses `virtual` or `override` keywords (mutually exclusive).

> ClassDefOverride: Use this one if your class definition use the c++ override keyword for virtual functions (If you use the regular ClassDef, the compiler might warn that the function declared in the ClassDef are missing the override keyword).

So the compiler warning this description refers to, must be https://clang.llvm.org/docs/DiagnosticsReference.html#winconsistent-missing-override ?!

Reply:
Do we have consensus on using `ClassDefOverride` once we modernize the `FairRootManager` class to use the `override` keyword? @ChristianTackeGSI Or do you have a different interpretation of when this macro shall be used?

Reply:
That sounds fine!

If we switch to "the override" world (make things `-Wsuggest-override`-clean), then we should also switch to `ClassDefOverride`.

As far, as I can tell, there is only one other place in this class that would get `override`:
```diff
-    Int_t Write(const char* name = 0, Int_t option = 0, Int_t bufsize = 0);
+    Int_t Write(const char* name = 0, Int_t option = 0, Int_t bufsize = 0) override;
```

All that said: This is only a suggestion. Do whatever fits the FairRoot world best!